--- libinklevel/Makefile.orig	Wed Dec  3 17:43:03 2003
+++ libinklevel/Makefile	Fri Dec  5 14:27:23 2003
@@ -1,18 +1,16 @@
-DESTDIR = /usr/local/
+#DESTDIR =
 
-CFLAGS = -Wall -O -fPIC -DPIC -I$(PWD)
+CFLAGS = $(OPTFLAGS) -Wall -fPIC -DPIC -I.
 
 all: libinklevel.so.2.6 
 
 install: all
-	mkdir -p $(DESTDIR)/include
-	install -m 644 -o root -g root inklevel.h $(DESTDIR)/include/inklevel.h
-	mkdir -p $(DESTDIR)/lib
-	install -m 755 -o root -g root libinklevel.so.2.6 \
-	$(DESTDIR)/lib/libinklevel.so.2.6
-	cd $(DESTDIR)/lib && rm -fr libinklevel.so && \
+	mkdir -p $(DESTDIR)/usr/include
+	install -c inklevel.h $(DESTDIR)/usr/include/inklevel.h
+	mkdir -p $(DESTDIR)/usr/lib
+	install -c libinklevel.so.2.6 $(DESTDIR)/usr/lib/libinklevel.so.2.6
+	cd $(DESTDIR)/usr/lib && rm -fr libinklevel.so && \
 	ln -s libinklevel.so.2.6 libinklevel.so
-	ldconfig $(DESTDIR)/lib
 
 clean:
 	rm -fr *.o *.so.2.6 *~ 
@@ -20,5 +18,5 @@
 libinklevel.o: libinklevel.c inklevel.h
 
 libinklevel.so.2.6: libinklevel.o 
-	gcc -shared -Wl,-soname,libinklevel.so.2 -o libinklevel.so.2.6 \
-	libinklevel.o -lc -lieee1284
+	$(CC) -shared -Wl,-soname,libinklevel.so.2 -o libinklevel.so.2.6 \
+	libinklevel.o -lieee1284
