--- libinklevel-0.7.3/Makefile.orig	2008-01-07 20:51:46.000000000 +0100
+++ libinklevel-0.7.3/Makefile	2008-08-25 21:11:00.086119225 +0200
@@ -1,6 +1,8 @@
 DESTDIR = 
 PREFIX = /usr/local
-CFLAGS += -Wall -O2 -fPIC -DPIC -I.
+LIBDIR = $(PREFIX)/lib
+OPTFLAGS = -O2
+CFLAGS += -Wall $(OPTFLAGS) -fPIC -DPIC -I.
 CC = gcc
 
 all: libinklevel.so.4.0.7.3
@@ -9,10 +11,10 @@
 	install -D -m 644 inklevel.h \
 		$(DESTDIR)/$(PREFIX)/include/inklevel.h
 	install -D -m 644 libinklevel.so.4.0.7.3 \
-		$(DESTDIR)/$(PREFIX)/lib/libinklevel.so.4.0.7.3
-	cd $(DESTDIR)/$(PREFIX)/lib && rm -fr libinklevel.so && \
+		$(DESTDIR)/$(LIBDIR)/libinklevel.so.4.0.7.3
+	cd $(DESTDIR)/$(LIBDIR) && rm -fr libinklevel.so && \
 		ln -s libinklevel.so.4.0.7.3 libinklevel.so 
-	cd $(DESTDIR)/$(PREFIX)/lib && rm -fr libinklevel.so.4 && \
+	cd $(DESTDIR)/$(LIBDIR) && rm -fr libinklevel.so.4 && \
 		ln -s libinklevel.so.4.0.7.3 libinklevel.so.4
 
 clean:
