diff -aurN libinklevel.orig/Makefile libinklevel/Makefile
--- libinklevel.orig/Makefile	2003-12-03 17:43:03.000000000 +0100
+++ libinklevel/Makefile	2003-12-04 17:10:08.000000000 +0100
@@ -1,18 +1,16 @@
-DESTDIR = /usr/local/
+#DESTDIR = /usr/local/
 
-CFLAGS = -Wall -O -fPIC -DPIC -I$(PWD)
+CFLAGS = $(OPTFLAGS) -Wall -O -fPIC -DPIC -I$(PWD)
 
 all: libinklevel.so.2.6 
 
 install: all
-	mkdir -p $(DESTDIR)/include
-	install -m 644 -o root -g root inklevel.h $(DESTDIR)/include/inklevel.h
-	mkdir -p $(DESTDIR)/lib
-	install -m 755 -o root -g root libinklevel.so.2.6 \
-	$(DESTDIR)/lib/libinklevel.so.2.6
-	cd $(DESTDIR)/lib && rm -fr libinklevel.so && \
+	mkdir -p $(DESTDIR)/usr/include
+	install -c inklevel.h $(DESTDIR)/usr/include/inklevel.h
+	mkdir -p $(DESTDIR)/usr/lib
+	install -c libinklevel.so.2.6 $(DESTDIR)/usr/lib/libinklevel.so.2.6
+	cd $(DESTDIR)/usr/lib && rm -fr libinklevel.so && \
 	ln -s libinklevel.so.2.6 libinklevel.so
-	ldconfig $(DESTDIR)/lib
 
 clean:
 	rm -fr *.o *.so.2.6 *~ 
